edx_rest_framework_extensions/tests/test_authentication.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+""" Tests for authentication classes. """
+
+import json
+
+import httpretty
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.test import override_settings, RequestFactory, TestCase
+from rest_framework.exceptions import AuthenticationFailed
+
+from edx_rest_framework_extensions.authentication import BearerAuthentication
+from edx_rest_framework_extensions.tests import factories
+
+OAUTH2_PROVIDER_URL = 'http://example.com/oauth2'
+
+User = get_user_model()
+
+
+class AccessTokenMixin(object):
+    """ Test mixin for dealing with OAuth2 access tokens. """
+    DEFAULT_TOKEN = 'abc123'
+
+    def mock_access_token_response(self, status=200, token=DEFAULT_TOKEN, username='fake-user'):
+        """ Mock the access token endpoint response of the OAuth2 provider. """
+        httpretty.register_uri(httpretty.GET, '{}/access_token/{}/'.format(OAUTH2_PROVIDER_URL, token),
+                               body=json.dumps({'username': username, 'scope': 'read', 'expires_in': 60}),
+                               content_type="application/json",
+                               status=status)
+
+
+@override_settings(OAUTH2_PROVIDER_URL=OAUTH2_PROVIDER_URL)
+class BearerAuthenticationTests(AccessTokenMixin, TestCase):
+    """ Tests for the BearerAuthentication class. """
+
+    def setUp(self):
+        super(BearerAuthenticationTests, self).setUp()
+        self.auth = BearerAuthentication()
+        self.factory = RequestFactory()
+
+    def create_authenticated_request(self, token=AccessTokenMixin.DEFAULT_TOKEN, token_name='Bearer'):
+        """ Returns a Request with the authorization set using the specified values. """
+        auth_header = '{token_name} {token}'.format(token_name=token_name, token=token)
+        request = self.factory.get('/', HTTP_AUTHORIZATION=auth_header)
+        return request
+
+    def test_authenticate_header(self):
+        """ The method should return the string Bearer. """
+        self.assertEqual(self.auth.authenticate_header(self.create_authenticated_request()), 'Bearer')
```

Comment:
As written this method doesn't need an authenticated request passed to it.  It is confusing to have all these references to a request being passed in if we don't really need it.  I'd prefer to see this not send any arguments in the test.